src/ui/pages/Login/SignIn.jsx

```diff
@@ -0,0 +1,217 @@
+import React, {useState} from 'react';
+import Avatar from '@mui/material/Avatar';
+import Button from '@mui/material/Button';
+import CssBaseline from '@mui/material/CssBaseline';
+import TextField from '@mui/material/TextField';
+
+import "./SignIn.css"
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import InputAdornment from '@mui/material/InputAdornment';
+
+import Box from '@mui/material/Box';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+import {ThemeProvider} from '@mui/material/styles';
+import {getCurrentTheme} from '../../themes/Theme';
+import {useDispatch} from "react-redux";
+import {useNavigate} from "react-router-dom";
+import {toast} from "react-toastify";
+import {useLoginMutation} from "../../../redux/features/api/authApi";
+import {setCredentials} from "../../../redux/features/state/authstate";
+
+
+const defaultTheme = getCurrentTheme();
+
+function SignIn() {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+
+    const [emailError, setEmailError] = useState(null);
+    const [passwordError, setPasswordError] = useState(null);
+
+    const [showPassword, setShowPassword] = useState(false);
+
+
+    const [login, {isLoading}] = useLoginMutation();
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+
+    if (isLoading) {
+        toast.loading("Logging in...", {
+            toastId: "loading",
+            autoClose: false
+        })
+    } else {
+        toast.dismiss("loading");
+    }
+
+    function validateEmail(email) {
+        if (!email) {
+            setEmailError("Email is required");
+            return false;
+        }
+        let pattern = "^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+.[a-zA-Z]{2,}$";
+        if (!email.match(pattern)) {
+            setEmailError("Invalid email");
+            return false;
+        }
+        return true;
+    }
+
+    function validatePassword(password) {
+        if (!password) {
+            setPasswordError("Password is required");
+            return false;
+        }
+        if (password.length < 6) {
+            setPasswordError("Password must be at least 6 characters");
+            return false;
+        }
+
+        if (!password.match("[a-zA-Z]") || !password.match("[0-9]")) {
+            setPasswordError("Password must contain letters and numbers");
+            return false;
+
+        }
+        return true;
+    }
```

Comment:
To make the component more cleaner , we might want to move it out

Reply:
As we discussed, the function also alters the props of the email and password fields. I suggest we refactor it later if we run into a scenario where we need this again.